### PR TITLE
fix uncaught exception in `WebSocketClient`

### PR DESCRIPTION
**Pull request by buntec**

### CODE REVIEW

**Comment by armanbilge** [comment]
Thanks! How was this manifesting? I'm wondering how easily we can add a test for it.

**Comment by buntec** [comment]
I noticed the uncaught `IllegalStateException` in the browser console (Firefox) when trying to connect to a valid WS url where no server was running (e.g., ws://localhost:9000/foo). The stack trace revealed that it was coming from `onclose`, which apparently is called *after* `onerror`. (`onmessage` is probably fine as is but doesn't hurt to remove the listener either).

**Comment by armanbilge** [comment]
Thanks! That sounds like something we can replicate in CI. Then maybe we can add a listener to the `window` `'error'` event and check that nothing is emitted.

File: dom/src/main/scala/org/http4s/dom/WebSocketClient.scala [hunk 1]
```diff
@@ -72,7 +72,11 @@ object WebSocketClient {
               cb(Right(ws))
             }
 
-            ws.onerror = e => cb(Left(js.JavaScriptException(e)))
+            ws.onerror = { e =>
+              ws.onmessage = null // prevent using dispatcher after shutdown
+              ws.onclose = null // ditto
+              cb(Left(js.JavaScriptException(e)))
+            }
```
Comment (armanbilge) [inline dom/src/main/scala/org/http4s/dom/WebSocketClient.scala:79]: So I did some more investigation and I found out the `'error'` handler is kind of pointless.
1. it doesn't provide any detail about the error, _by-design_
    https://websockets.spec.whatwg.org/#eventdef-websocket-error
2. it's always followed by a `'close'` event (also by spec), so we might as well just use that

along these lines, I have a different proposal in 86bfb39e83d9b9ba93ace410961ae5e8d0a87860, wdyt?

Comment (buntec) [reply dom/src/main/scala/org/http4s/dom/WebSocketClient.scala:79]: Great, one less callback to worry about! How shall we proceed?

**Comment by buntec** [comment]
Superseded by #267.

**Pull request closed without merging**